sklearn/linear_model/ridge.py

```diff
@@ -583,20 +583,24 @@ def fit(self, X, y):
         y : array-like, shape = [n_samples]
             Target values
 
+        sample_weight : float or numpy array of shape (n_samples,)
+            Sample weight.
+
         Returns
         -------
         self : returns an instance of self.
         """
+        if sample_weight is None:
+            sample_weight = 1.
```

Comment:
Could you also modify this line:
https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/linear_model/ridge.py#L285
so that `has_sw` is `False` if `sample_weight == 1.0`.

`sample_weight` is currently implemented with a rescaling of the data. We can avoid copying the data in the case `sample_weight == 1.0`.

Another option is to check for `sample_weight == 1.0` directly in `_rescale_data`.